File: tree_py/binarytreetest/testbst01.py
from binarytree import BinarySearchTree


def dotask():
    bst = BinarySearchTree(candd_removal='right')

    for value in [12, 39, 20, 7, 26, 45, 19, 8]:
        bst.insert(value)

    print(bst)

    print('count:', bst.count())
    print('min:', bst.min())
    print('max:', bst.max())
    print('get key:', bst.get(20))

    print('Using for each to iterate nodes in tree')

    for value in bst:
        print(value, end='  ')

    print(end='\n')

    _ = bst.remove(800)
    _ = bst.remove(12)

    print(bst)
    print('count:', bst.count())

File: bintree_csharp/test/TestBst01.cs
using System;
using my.binarytree;


namespace my.test
{
    class TestBst01 : TestBase
    {
        public override void doTask()
        {
            var bst = new BinSearchTree<int, BinNodeConcrete<int>>();

            foreach (var value in new int[] { 12, 39, 20, 7, 26, 45, 19, 8 })
                bst.Insert(value);

            Console.WriteLine("\n count: " + bst.Count);
            Console.WriteLine("\n min: " + bst.Min());
            Console.WriteLine("\n max: " + bst.Max());
            Console.WriteLine("\n contain: " + bst.Contain(20));
            Console.WriteLine("\n height: " + bst.Height);

            Console.WriteLine("\n print tree: ");
            PrintTree(bst);


            Console.WriteLine("\n");


            bst.Remove(800);
            bst.Remove(12);

            Console.WriteLine("\n count: " + bst.Count);
            Console.WriteLine("\n height: " + bst.Height);

            Console.WriteLine("\n print tree: ");
            PrintTree(bst);


            Console.WriteLine();
        }
    }
}
